tests/test_manager.py

```diff
@@ -617,3 +618,38 @@ def test_db_journal_mode(any_fid_manager_class, fid_db_path, jp_root_dir, db_jou
         cursor = fid_manager.con.execute("PRAGMA journal_mode")
         actual_journal_mode = cursor.fetchone()
         assert actual_journal_mode[0].upper() == expected_journal_mode
+
+
+# This test demonstrates an issue raised in
+# https://github.com/jupyter-server/jupyter_server_fileid/pull/76
+# which was later fixed in
+# https://github.com/jupyter-server/jupyter_server_fileid/pull/77
+#
+# We use this unit test to catch this edge case and ensure
+# its covered going forward.
+def test_multiple_fileIdManager_connections_after_exception(fid_db_path):
+    original_file_path = "/path/to/file"
+    copy_location = "/path/to/copy"
+    another_copy_location = "/path/to/other"
+
+    # Setup an initial file ID manager connected to a sqlite database.
+    manager_1 = ArbitraryFileIdManager(db_path=fid_db_path)
+
+    # Create an initial ID for this file
+    manager_1.index(original_file_path)
+    # Copy the file
+    manager_1.copy(original_file_path, copy_location)
+    # Try copying the file again.
+    try:
+        manager_1.copy(original_file_path, copy_location)
+    # We expect this to fail because the file is already in the database.
+    except sqlite3.IntegrityError:
+        pass
```

Comment:
I don't think this `try/except` block does anything in the test, since all exceptions are being caught in the File ID manager. So the `sqlite3.IntegrityError` exception is not raised in the second call to `copy()`, even though it is invalid.

Reply:
If you accept the proposal I offered in my review, it would be nice to modify this test to _only pass_ if an exception was raised using some kind of flag, e.g.

```suggestion
    excepted = False
    try:
        manager_1.copy(original_file_path, copy_location)
    # We expect this to fail because the file is already in the database.
    except sqlite3.IntegrityError:
        excepted = True
        pass
   
   if not excepted: ... # fail the test
```

This would ensure public methods still raise an exception for invalid operations. 🤔